ethers/providers/jsonrpc/subscriptions.nim

```diff
@@ -164,8 +164,14 @@ proc new*(_: type JsonRpcSubscriptions,
       raise e
     except CatchableError as e:
       if "filter not found" in e.msg:
-        let filter = subscriptions.filters[originalId]
-        let newId = await subscriptions.client.eth_newFilter(filter)
+        var newId: JsonNode
+        # If there exists filter for given ID, then the filter was a log filter
+        # otherwise it was a block filter
+        if subscriptions.filters.hasKey(originalId):
+          let filter = subscriptions.filters[originalId]
+          newId = await subscriptions.client.eth_newFilter(filter)
+        else:
```

Comment:
This seems loose... maybe we could store an object in `subscriptionMapping` that has an `id` and `kind`, and then here we can check the `kind` before deciding what to do...

Reply:
I was also thinking a bit about this "looseness", but I did not really find any reason why it should lead to bug. 

If there is a callback (and hence, that is why this function is called), then the callback had to be created either through `subscribeLogs,` and hence it will have an entry in `filters` or through `suscribeBlocks` and then it will not have an entry in `filters`. IMHO this is good enough reasoning. If you think there will "magically disappear filters" then we have a problem and a bug somewhere anyway and will have to track that case down...

Soooo I would leave it as is.

Reply:
> If you think there will "magically disappear filters" then we have a problem and a bug somewhere anyway and will have to track that case down...

It's more about if `filters` functionality is changed at some point in the future, then we may introduce a bug that isn't easy to spot/debug.

It's also not the most easy to reason about... As I was reading the change, i thought "why wouldn't `filters` also contain a *block* filter?" Then I had to look at how creating a block filter doesn't add to `filters`. 

It seems a bit fragile honestly.

Reply:
Alright, so what about renaming `filters` to `logFilters` in order to better communicate it? Because that is what that really is. In the context of this Table, there are no "block filters", because block filters are only subscriptions for new blocks, not for "notify me when some specific blocks arrive" and hence it does not really make sense to store them in that Table...

Reply:
Yep, that works 👍

Reply:
> there are no "block filters"

Might be worth updating the comment?